include/taco/taco_tensor_t.h

```diff
@@ -22,6 +22,27 @@ typedef struct taco_tensor_t {
   int32_t      vals_size;     // values array size
 } taco_tensor_t;
 
+int taco_gallop(int *array, int arrayStart, int arrayEnd, int target) {
```

Comment:
@rohany I had to add this for the cli test to pass. I also noticed that this file is missing the `taco_binarySearch` functions, among possibly some other stuff in codegen_c.cpp. Should they be added too?

Reply:
I don't believe you should need to include that there. What errors do you get without it?

Reply:
This one: 
```
 ✗ test -s=mergeby
   (from function `tacocompile' in file test/taco-cli-test.bats, line 154,
    in test file test/taco-cli-test.bats, line 229)
     `tacocompile "$expression" $rformat $sformat $tformat "$directive"' failed
   trying: taco 'q(i,j,k) = r(i,j)*s(j,k)*t(i,k)' '-f=r:ss' '-f=s:ss' '-f=t:ss' '-s=mergeby(j,Gallop)'
   compile command: /usr/bin/gcc -O2 -Wall -pedantic -o /var/folders/d9/zp4vpq2147zcykcc957fq78w0000gn/T/taco_test_program.exe /var/folders/d9/zp4vpq2147zcykcc957fq78w0000gn/T/taco_test_program.c
   compiler returned 1
   output: /var/folders/d9/zp4vpq2147zcykcc957fq78w0000gn/T/taco_test_program.c:80:14: error: implicit declaration of function 'taco_gallop' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
           jr = taco_gallop(r2_crd, jr, r2_pos[(ir + 1)], j);
                ^
   1 error generated.
```
which was fixed by adding `taco_gallop` here.

Reply:
Does the generated code in `/var/folders/d9/zp4vpq2147zcykcc957fq78w0000gn/T/taco_test_program.c` have auxiliary function definitions (like binary search and gallop) at the top? Or does it only have header includes?

Reply:
It doesn't have the auxiliary functions (attached). I just noticed there's also some boiler plate code [here](https://github.com/tensor-compiler/taco/blob/4c24193f8da3883693710e786f4ede5371b14e71/test/taco-cli-test.bats#L134), maybe this is a better place to put gallop?
[taco_test_program.c.txt](https://github.com/tensor-compiler/taco/files/8721095/taco_test_program.c.txt)

Reply:
I see. This CLI testing infrastructure is written in a slightly unfortunate way... I think leaving a TODO with a pointer to this discussion about adding a CLI test is probably fine, since pasting the gallop code in there as well is not great.

Either way, I think adding a gtest that fills a tensor with random coordinates and checks that the result with and without galloping are equal is a fine test.